provider/reprovider.go

```diff
@@ -114,6 +118,8 @@ func New(ds datastore.Batching, opts ...Option) (System, error) {
 		maxReprovideBatchSize: math.MaxUint,
 		provideWorkerCount:    defaultProvideWorkerCount,
 		keyPrefix:             DefaultKeyPrefix,
+		provideCounter:        metrics.New("ipfs.boxo.provider.provideCount", "Number of provides since node is running").Counter(),
+		reprovideCounter:      metrics.New("ipfs.boxo.provider.reprovideCount", "Number of reprovides since node is running").Counter(),
```

Comment:
Regarding the metric name... I believe all metrics in boxo use `metrics.NewCtx` and figure out their metric prefix from a context.

In HTTP I hardcoded that prefix though, which probably wasn't intelligent: https://github.com/ipfs/boxo/blob/67939861ceb726ed6f97474eed381c89b1be8bd9/bitswap/network/httpnet/metrics.go#L75

If you can get a context, then it doesn't hurt using it. And otherwise, the metric name would, based on prior art, be `provider_provide_count` and `reprovider_reprovide_count`.

(I think, I'm not very sure about this. I am more sure that the `ipfs.boxo` prefix is probably not part of any existing metric).